presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -195,4 +246,76 @@ private void initializeColumnReaders()
             columnReadersMap.put(column, ParquetColumnReader.createReader(column));
         }
     }
+
+    private Block readBlock(String name, Type type, List<String> path, List<Integer> offsets, List<Integer> positions)
```

Comment:
`offsets` is called `lengths` in some parts of the code (e.g., in `readMap` and `readArray`), we should stick to one of these (offset is better I think).